### PR TITLE
Implement metadata api

### PR DESCRIPTION
**Pull request by vladoatanasov**
TODO:
- [x] Unit tests

* Protocol files were generated
* The metadata API returns mock data to satisfy the Kafka protocol for now. This will be updated in the future

### CODE REVIEW

**Comment by PawelK2012** [comment]
@vladoatanasov 108 files  to review?  We need involve HR 😂 https://github.com/PlaytikaOSS/small-pull-request-manifesto

File: protocol/prep_encoder.go [hunk 1]
```diff
@@ -41,7 +41,8 @@ func (pe *prepEncoder) putInt64(in int64) {
 }
 
 func (pe *prepEncoder) putUUID(in uuid.UUID) error {
-	return errors.New("not implemented")
+	pe.length += 16
+	return nil
```
Comment (doaonduty) [inline protocol/prep_encoder.go:45]: @vladoatanasov  is this related to the well kept secret header size that we talked earlier sir?

File: server.go [hunk 2]
```diff
@@ -79,33 +81,56 @@ func (client *Client) handleRequest() {
 		// encoded := hex.EncodeToString(messageBytes)
 		// fmt.Println(encoded)
 
+		// We parse the header twice, first time parse only API key and API version, from which we can
+		// infer the correct header version and then parse that again in the API code to get the full header.
 		header := &protocol.RequestHeader{}
-		protocol.VersionedDecode(messageBytes, header, 2)
+		protocol.VersionedDecode(messageBytes, header, 1)
```
Comment (doaonduty) [inline server.go:87]: @vladoatanasov thank you for putting clear comments.

**Review by doaonduty — APPROVED** [review]

File: protocol/prep_encoder.go (hunk 1 above)
Comment (vladoatanasov) [reply protocol/prep_encoder.go:45]: no, this function encodes a UUID type into the Kafka protocol, which encodes to a 16 byte code. https://kafka.apache.org/protocol.html#protocol_types

**Pull request merged**